udf/kv_array_append_to_json_string.sql

```diff
@@ -0,0 +1,35 @@
+/*
+
+Returns a JSON string which has the `pair` appended to the provided `input` JSON string.
+
+Example:
+  udf_kv_array_append_to_json_string('{"foo":"bar"}', [STRUCT("baz" AS key, "boo" AS value)])
+
+  '{"foo":"bar","baz":"boo"}'
+
+  udf_kv_array_append_to_json_string('{}', [STRUCT("baz" AS key, "boo" AS value)])
+
+  '{"baz": "boo"}'
+
+*/
+CREATE OR REPLACE FUNCTION udf.kv_array_append_to_json_string(input STRING, arr ANY TYPE) AS (
+  CONCAT(
+    RTRIM(input, "}"),
+    IF(input = "{}", "", ","),
+    TRIM(`moz-fx-data-shared-prod.udf.kv_array_to_json_string`(arr), "{")
```

Comment:
```suggestion
    TRIM(udf.kv_array_to_json_string(arr), "{")
```

Within the UDF definitions, we do some substitutions in testing, so prefer non-fully-qualified UDF refs. View definitions, on the other hand, are required by BQ to use fully-qualified references, so you're doing the right thing there.